src/compiler/parser.ts

```diff
@@ -7613,13 +7615,17 @@ namespace ts {
                 case "reference": {
                     const referencedFiles = context.referencedFiles;
                     const typeReferenceDirectives = context.typeReferenceDirectives;
+                    const libReferenceDirectives = context.libReferenceDirectives;
                     forEach(toArray(entryOrList), (arg: PragmaPsuedoMap["reference"]) => {
                         if (arg.arguments["no-default-lib"]) {
                             context.hasNoDefaultLib = true;
                         }
                         else if (arg.arguments.types) {
                             typeReferenceDirectives.push({ pos: arg.arguments.types.pos, end: arg.arguments.types.end, fileName: arg.arguments.types.value });
                         }
+                        else if (arg.arguments.lib) {

```

Comment:
#23904 Could avoid repeated code here

Reply:
Do you want me to do this or do you want to do in your other PR once this is merged?

Reply:
I'll close the other PR since it doesn't have much benefit with only 2 branches, but maybe it would have a benefit with 3 branches, or there's a better way to share the code.